Drupal/modules/custom/custom_contact_api/src/Plugin/rest/resource/ContactFormResource.php

```diff
@@ -0,0 +1,131 @@
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param array $serializer_formats
+   *   The available serialization formats.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The currently authenticated user.
+   * @param \Drupal\contact\MailHandlerInterface $mail_handler
+   *   The contact mail handler service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, $serializer_formats, LoggerInterface $logger, EntityTypeManagerInterface $entity_type_manager, AccountInterface $current_user, MailHandlerInterface $mail_handler)
+  {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $serializer_formats, $logger);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->currentUser = $current_user;
+    $this->mailHandler = $mail_handler;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition)
+  {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->getParameter('serializer.formats'),
+      $container->get('logger.factory')->get('rest'),
+      $container->get('entity_type.manager'),
+      $container->get('current_user'),
+      $container->get('contact.mail_handler'),
+    );
+  }
+
+  /**
+   * Responds to POST requests.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *  Represents an HTTP request.
+   *
+   * @return ResourceResponse
+   */
+  public function post(Request $request) {
+    $data = json_decode($request->getContent(), TRUE);
+
+    // For invalid inputs respond with cannot process your request message.
+    if (empty($data['subject']) || empty($data['message'])) {
+      throw new HttpException(422, 'Invalid input, either subject or message missing.');
+    }
+    if (!$this->entityTypeManager->getStorage('user')->load($data['recipient'])) {
+      throw new HttpException(422, 'Invalid input, recipient does not exists.');
+    }
+
+    // Create the message entity using the contact form.
+    $message = Message::create([
+      'contact_form' => 'personal',
+      'name' => $this->currentUser->getAccountName(),
+      'mail' => $this->currentUser->getEmail(),
+      'message' => $data['message'],
+      'subject' => $data['subject'],
+      'recipient' => $data['recipient'], // Recipient's user-id.
+      'copy' =>  isset($data['copy']) ? 1 : 0, // Send yourself a copy of the email.
+    ]);
+    $message->save();
+
+    // Send Email.
+    $this->mailHandler->sendMailMessages($message, $this->currentUser);
+
+    return new ResourceResponse(['message' => 'Contact form submitted successfully']);
```

Comment:
To handle cases where the API response is not 200 (OK), and to inform the user that the email has not been sent, you can modify the post method in the ContactFormResource class to include error handling for unsuccessful API responses.

Reply:
for example `public function post(Request $request) {
    $data = json_decode($request->getContent(), TRUE);

    // For invalid inputs respond with cannot process your request message.
    if (empty($data['subject']) || empty($data['message'])) {
        throw new HttpException(Response::HTTP_UNPROCESSABLE_ENTITY, 'Invalid input, either subject or message missing.');
    }
    if (!$this->entityTypeManager->getStorage('user')->load($data['recipient'])) {
        throw new HttpException(Response::HTTP_UNPROCESSABLE_ENTITY, 'Invalid input, recipient does not exist.');
    }

    // Create the message entity using the contact form.
    $message = Message::create([
        'contact_form' => 'personal',
        'name' => $this->currentUser->getAccountName(),
        'mail' => $this->currentUser->getEmail(),
        'message' => $data['message'],
        'subject' => $data['subject'],
        'recipient' => $data['recipient'], // Recipient's user-id.
        'copy' =>  isset($data['copy']) ? 1 : 0, // Send yourself a copy of the email.
    ]);
    $message->save();

    // Send Email.
    try {
        $this->mailHandler->sendMailMessages($message, $this->currentUser);
    } catch (\Exception $e) {
        // If an error occurs while sending email, inform the user.
        throw new HttpException(Response::HTTP_INTERNAL_SERVER_ERROR, 'Email could not be sent. Please try again later.');
    }

    return new ResourceResponse(['message' => 'Contact form submitted successfully']);
}`

Reply:
Hi @lomasgupta thank you, this is a very good suggestion. But I am just trying to understand - should we care about the exceptions such as database unavailable while making DB queries? Or for the current situation of sending email and enclosing our code in try-catch? As I am seeing even the main form builder code for /user/{uid}/contact (web/core/modules/contact/src/MessageForm.php) does not do it. Any thoughts?

Reply:
Hi @axelabhay It's just a suggestion, as I've also included a comment: "// If an error occurs while sending email, inform the user." This would make it more engaging for the end user.

Reply:
Perfect! I Will update the code shortly. Thank you.

Reply:
Done.